### PR TITLE
Add Kani tooling and local smoke targets (4.1.1)

### PR DESCRIPTION
**Pull request by leynos**
## Summary

This branch implements roadmap task (4.1.1), adding the pinned local Kani tooling contract for Netsuke's formal-verification workflow.

It adds `tools/kani/VERSION`, `scripts/install-kani.sh`, and the local `make kani`, `make kani-full`, and `make formal-pr` targets. It also documents the developer workflow and marks the roadmap item complete after validation.

Roadmap task: (4.1.1)
Execplan: [docs/execplans/4-1-1-kani-tooling-and-local-smoke-targets.md](https://github.com/leynos/netsuke/blob/4-1-1-kani-tooling-and-local-smoke-targets/docs/execplans/4-1-1-kani-tooling-and-local-smoke-targets.md)

## Review walkthrough

- Start with the ExecPlan to review the completed decisions, validation evidence, and scope boundary.
- Review `tools/kani/VERSION` and `scripts/install-kani.sh` for the pinned `kani-verifier` installation path.
- Review `Makefile` for the opt-in Kani smoke, full-suite, and formal pull-request targets.
- Review `docs/developers-guide.md` and `docs/roadmap.md` for the developer workflow and completed roadmap checkboxes.

## Validation

- `scripts/install-kani.sh`: passed, installed `kani-verifier` 0.67.0 and ran `cargo kani setup`
- `make kani`: passed
- `make formal-pr`: passed
- `make kani-full`: passed; currently reports that no `#[kani::proof]` harnesses exist
- `mbake validate Makefile`: passed
- `make check-fmt`: passed
- `make lint`: passed
- `make test`: passed
- `make markdownlint`: passed
- `make nixie`: passed
- `coderabbit review --agent`: passed with 0 findings on each milestone and final run

## Notes

- `make fmt` was attempted, but it still fails on pre-existing Markdown line-length and table issues in unrelated documentation. Formatter-only edits outside this branch's intended files were restored.
- This PR intentionally does not add CI jobs, proof harnesses, Proptest coverage, or Verus proofs; those remain deferred to later roadmap items.

## Summary by Sourcery

Add pinned Kani formal-verification tooling and local make targets for smoke and full runs, along with developer documentation and roadmap updates.

New Features:
- Introduce pinned Kani version metadata under tools/kani/VERSION for formal verification tooling.
- Add a Bash installer script to install and set up the repository-supported kani-verifier version.
- Add make targets for Kani smoke checks, full verification runs, and a formal pull-request path.

Enhancements:
- Document the Kani-based formal-verification workflow in the developer guide.
- Mark roadmap item 4.1.1 and its subitems as completed and add a detailed execution plan document for Kani tooling and local smoke targets.

Documentation:
- Add an execution plan document describing the design, constraints, risks, and outcomes for Kani tooling and local smoke targets.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,6 +44,15 @@ markdownlint: ## Lint Markdown files
 nixie: ## Validate Mermaid diagrams
 	nixie --no-sandbox
 
+kani: ## Run the Kani local smoke check
+	$(KANI) $(KANI_SMOKE_FLAGS)
```
[the remark this answers is not in the document]
Comment (leynos) [reply Makefile]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai please create a new GitHub issue for the following:

Note that tests for shell scripts must be written in Python using `pytest` and `pytest-bdd` (where appropriate) using https://github.com/leynos/cuprum for shell execution and https://github.com/leynos/cmd-mox for command mocking

### ❌ Failed checks (2 errors, 2 warnings, 1 inconclusive)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | Substantial functionality added (installer, version checks, targets) with only manual smoke validation. No automated tests prevent future regressions or catch broken implementations. | Add tests for install-kani.sh (version format, missing files), check-kani-version.sh (version mismatch detection), and Makefile targets via rstest-bdd or shell integration tests. |
| Module-Level Documentation | ❌ Error | New Bash scripts contain only single-line doc comments. Custom check requires comprehensive docstrings explaining purpose, utility, function, and relationships to other components. | Expand module-level documentation in scripts/install-kani.sh and scripts/check-kani-version.sh with multi-line comments covering purpose, utility, function, parameters, and Makefile integration. |
| Docstring Coverage | ⚠️ Warning | Docstring coverage is 0.00% which is insufficient. The required threshold is 80.00%. | Write docstrings for the functions missing them to satisfy the coverage threshold. |
| Testing (Unit And Behavioural) | ⚠️ Warning | PR adds Makefile targets and shell scripts for command-line execution but provides no tests for them. Custom check requires tests for system-level behaviour changes. | Add tests covering shell scripts' error paths, edge cases, and invariants. Test Makefile targets. Use shell testing or Rust integration tests. |
